app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -41,19 +43,23 @@ def ems_inv_to_hashes
 
       private
 
+      def get_resource_groups
+        resource_groups = @rgs.list.select do |resource_group|
+          resource_group.location == @ems.provider_region
+        end
+        process_collection(resource_groups, :resource_groups) do |resource_group|
+          parse_resource_group(resource_group)
+        end
+      end
+
       def get_series
         series = []
-        get_locations.each do |location|
-          begin
-            series << @vmm.series(location)
-          rescue ::Azure::Armrest::BadGatewayException,
-                 ::Azure::Armrest::GatewayTimeoutException,
-                 ::Azure::Armrest::BadRequestException
-            next
-          end
+        begin
+          series = @vmm.series(@ems.provider_region)
+        rescue ::Azure::Armrest::BadGatewayException,
+               ::Azure::Armrest::GatewayTimeoutException,
+               ::Azure::Armrest::BadRequestException
```

Comment:
This should probably log a warning that it couldn't pick up the flavors for the particular Azure region.